src/controllers/commands/CommandController.cpp

```diff
@@ -1725,6 +1725,119 @@ void CommandController::initialize(Settings &, Paths &paths)
+                    errorMessage += "You don't have permission to "
+                                    "perform that action.";
+                }
+                break;
+
+                case Error::Ratelimited: {
+                    errorMessage += "You are being ratelimited by Twitch. Try "
+                                    "again in a few seconds.";
+                }
+                break;
+
+                case Error::Forwarded: {
+                    errorMessage = message;
+                }
+                break;
+
+                case Error::Unknown:
+                default: {
+                    errorMessage += "An unknown error has occurred.";
+                }
+                break;
+            }
+            return errorMessage;
+        };
+    this->registerCommand("/emoteonly", [formatChatSettingsError](
+                                            const QStringList & /* words */,
+                                            auto channel) {
+        auto currentUser = getApp()->accounts->twitch.getCurrent();
+        if (currentUser->isAnon())
+        {
+            channel->addMessage(makeSystemMessage(
+                "You must be logged in to update chat settings!"));
+            return "";
+        }
+        auto *twitchChannel = dynamic_cast<TwitchChannel *>(channel.get());
+        if (twitchChannel == nullptr)
+        {
+            channel->addMessage(makeSystemMessage(
+                "The /emoteonly command only works in Twitch channels"));
+            return "";
+        }
+
+        if (twitchChannel->accessRoomModes()->emoteOnly)
+        {
+            channel->addMessage(
+                makeSystemMessage("This room is already in emote-only mode."));
+            return "";
+        }
+
+        getHelix()->updateEmoteMode(
+            twitchChannel->roomId(), currentUser->getUserId(), true,
+            [](auto) {
+                //we'll get a message from irc
+            },
+            [channel, formatChatSettingsError](auto error, auto message) {
+                channel->addMessage(
+                    makeSystemMessage(formatChatSettingsError(error, message)));
+            });
+        return "";
+    });
+    this->registerCommand(
+        "/emoteonlyoff", [formatChatSettingsError](
+                             const QStringList & /* words */, auto channel) {
+            auto currentUser = getApp()->accounts->twitch.getCurrent();
+            if (currentUser->isAnon())
+            {
+                channel->addMessage(makeSystemMessage(
+                    "You must be logged in to update chat settings!"));
+                return "";
+            }
+            auto *twitchChannel = dynamic_cast<TwitchChannel *>(channel.get());
+            if (twitchChannel == nullptr)
+            {
+                channel->addMessage(makeSystemMessage(
+                    "The /emoteonlyoff command only works in Twitch channels"));
+                return "";
```

Comment:
This is almost the same as in other twitch commands. In my opinion you should be able to do something like this:

```cpp
this->registerTwitchCommand("/xd", 
                            [](const QStringList&,
                               std::shared_ptr<TwitchChannel>,
                               std::shared_ptr<TwitchAccount>) { 
// ...
});
```